### ISSUE TITLE
Rendering delay in Africa?

### ISSUE THREAD

**Issue by sommerluk** [openstreetmap/operations#625]
The path https://www.openstreetmap.org/way/1062402274#map=18/5.36463/-4.04496 has been created three days ago but still does not appear on the standard layer at https://www.openstreetmap.org/#map=17/5.36519/-4.04598 (even on high zoom levels) when accessing this from Ivory Coast.

During the last months, I've often seen that the tiles are very outdated, often a couple of days, and sometimes weeks, when accessing them from Ivory Coast. Reloading in the browser (F5) does not help, adding “/diry” does not help either. However, when I access the same URL via the Opera browser using its built-in VPN (which probably has a location outside Africa?), I get up-to-date tiles immediately. So I suppose this is a problem of the cache server?

Is there a systematic reason that the cache server accessible from Ivory Coast is much more outdated than the cache servers accessible from Europe?

**Comment by tomhughes** [comment]
We don't run the cache servers so we have no way to tell, but a shift reload will bust that cache anyway if you're on www.openstreetmap.org so we should be able to rule that out.

Is it possible that the ISP you're using has some sort of transparent proxy cache they are forcing you through?

**Comment by sommerluk** [comment]
I’ve created now created a new user name on my Linux machine and tested viewing the map again from this user name (so no old browser cache) and the problem persists. So I think the browser cache is not the problem.

I've tested this with two different ISP: Orange and MTN (out of four ISP available in Ivory Coast). Same result. 

I suspect there might be some sort of proxy, because Wikipedia blocks me from editing when using Orange as ISP, and the error message is “blocked proxy server”…

**Comment by tomhughes** [comment]
Is there some sort of government level national firewall that everything is being forced through?

**Comment by sommerluk** [comment]
IPS Orange:


tile.openstreetmap.org debug
Server Stats

Render Server: pyrene.openstreetmap.org
Load Average: 64.22
File Status

Browser Request Headers

CONTEXT_DOCUMENT_ROOT: /srv/tile.openstreetmap.org/cgi-bin/
CONTEXT_PREFIX: /cgi-bin/
DOCUMENT_ROOT: /srv/tile.openstreetmap.org/html
GATEWAY_INTERFACE: CGI/1.1
HTTPS: on
HTTP_ACCEPT: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,*/*;q=0.8
HTTP_ACCEPT_ENCODING: gzip
HTTP_ACCEPT_LANGUAGE: de-DE,de;q=0.9,es-ES;q=0.8,es;q=0.6,en-GB;q=0.5,en;q=0.4,fr-FR;q=0.3,fr;q=0.1
HTTP_CDN_LOOP: Fastly, Fastly
HTTP_DNT: 1
HTTP_FASTLY_CLIENT: 1
HTTP_FASTLY_FF: ovFvuRB7uwtjLzzSaXilgD4YOKcBKgU1q8cIkdUMkC0=!YYZ!cache-yyz4548-YYZ, ovFvuRB7uwtjLzzSaXilgD4YOKcBKgU1q8cIkdUMkC0=!YYZ!cache-yyz4531-YYZ
HTTP_FASTLY_ORIG_ACCEPT_ENCODING: gzip, deflate, br
HTTP_FASTLY_SSL: 1
HTTP_HOST: tile.openstreetmap.org
HTTP_REFERER: https://mail.google.com/
HTTP_SEC_FETCH_DEST: document
HTTP_SEC_FETCH_MODE: navigate
HTTP_SEC_FETCH_SITE: cross-site
HTTP_SEC_FETCH_USER: ?1
HTTP_UPGRADE_INSECURE_REQUESTS: 1
HTTP_USER_AGENT: Mozilla/5.0 (X11; Linux x86_64; rv:100.0) Gecko/20100101 Firefox/100.0
HTTP_X_FORWARDED_FOR: 160.154.248.187
HTTP_X_FORWARDED_HOST: tile.openstreetmap.org
HTTP_X_FORWARDED_SERVER: cache-yyz4548-YYZ
HTTP_X_TIMER: S1653402018.771803,VS0
HTTP_X_VARNISH: 232305047, 902209984
LC_CTYPE: C.UTF-8
PATH: /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin
QUERY_STRING:
REMOTE_ADDR: 160.154.248.187
REMOTE_PORT: 5260
REQUEST_METHOD: GET
REQUEST_SCHEME: https
REQUEST_URI: /cgi-bin/debug
SCRIPT_FILENAME: /srv/tile.openstreetmap.org/cgi-bin/debug
SCRIPT_NAME: /cgi-bin/debug
SCRIPT_URI: https://tile.openstreetmap.org/cgi-bin/debug
SCRIPT_URL: /cgi-bin/debug
SERVER_ADDR: 140.211.167.98
SERVER_ADMIN: webmaster@openstreetmap.org
SERVER_NAME: tile.openstreetmap.org
SERVER_PORT: 443
SERVER_PROTOCOL: HTTP/1.1
SERVER_SIGNATURE:
Apache/2.4.41 (Ubuntu) Server at tile.openstreetmap.org Port 443

SERVER_SOFTWARE: Apache/2.4.41 (Ubuntu)
SSL_TLS_SNI: render.openstreetmap.org

**Comment by sommerluk** [comment]
ISP MTN:


tile.openstreetmap.org debug
Server Stats

Render Server: odin.openstreetmap.org
Load Average: 19.02
File Status

Browser Request Headers

CONTEXT_DOCUMENT_ROOT: /srv/tile.openstreetmap.org/cgi-bin/
CONTEXT_PREFIX: /cgi-bin/
DOCUMENT_ROOT: /srv/tile.openstreetmap.org/html
GATEWAY_INTERFACE: CGI/1.1
HTTPS: on
HTTP_ACCEPT: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,*/*;q=0.8
HTTP_ACCEPT_ENCODING: gzip
HTTP_ACCEPT_LANGUAGE: de-DE,de;q=0.9,es-ES;q=0.8,es;q=0.6,en-GB;q=0.5,en;q=0.4,fr-FR;q=0.3,fr;q=0.1
HTTP_CDN_LOOP: Fastly, Fastly
HTTP_DNT: 1
HTTP_EARLY_DATA: 1
HTTP_FASTLY_CLIENT: 1
HTTP_FASTLY_FF: ovFvuRB7uwtjLzzSaXilgD4YOKcBKgU1q8cIkdUMkC0=!LCY!cache-lcy19262-LCY, ovFvuRB7uwtjLzzSaXilgD4YOKcBKgU1q8cIkdUMkC0=!LCY!cache-lcy19252-LCY
HTTP_FASTLY_ORIG_ACCEPT_ENCODING: gzip, deflate, br
HTTP_FASTLY_SSL: 1
HTTP_HOST: tile.openstreetmap.org
HTTP_SEC_FETCH_DEST: document
HTTP_SEC_FETCH_MODE: navigate
HTTP_SEC_FETCH_SITE: none
HTTP_SEC_FETCH_USER: ?1
HTTP_UPGRADE_INSECURE_REQUESTS: 1
HTTP_USER_AGENT: Mozilla/5.0 (X11; Linux x86_64; rv:100.0) Gecko/20100101 Firefox/100.0
HTTP_X_FORWARDED_FOR: 105.235.71.143
HTTP_X_FORWARDED_HOST: tile.openstreetmap.org
HTTP_X_FORWARDED_SERVER: cache-lcy19262-LCY
HTTP_X_TIMER: S1653402126.440120,VS0
HTTP_X_VARNISH: 952238587, 735833176
LC_CTYPE: C.UTF-8
PATH: /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
QUERY_STRING:
REMOTE_ADDR: 105.235.71.143
REMOTE_PORT: 37140
REQUEST_METHOD: GET
REQUEST_SCHEME: https
REQUEST_URI: /cgi-bin/debug
SCRIPT_FILENAME: /srv/tile.openstreetmap.org/cgi-bin/debug
SCRIPT_NAME: /cgi-bin/debug
SCRIPT_URI: https://tile.openstreetmap.org/cgi-bin/debug
SCRIPT_URL: /cgi-bin/debug
SERVER_ADDR: 130.117.76.15
SERVER_ADMIN: webmaster@openstreetmap.org
SERVER_NAME: tile.openstreetmap.org
SERVER_PORT: 443
SERVER_PROTOCOL: HTTP/1.1
SERVER_SIGNATURE:
Apache/2.4.41 (Ubuntu) Server at tile.openstreetmap.org Port 443

SERVER_SOFTWARE: Apache/2.4.41 (Ubuntu)
SSL_TLS_SNI: render.openstreetmap.org

**Comment by sommerluk** [comment]
ISP Orange through Opera with built-in VPN:

tile.openstreetmap.org debug
Server Stats
Render Server: pyrene.openstreetmap.org
Load Average: 53.07

File Status
Browser Request Headers
CONTEXT_DOCUMENT_ROOT: /srv/tile.openstreetmap.org/cgi-bin/
CONTEXT_PREFIX: /cgi-bin/
DOCUMENT_ROOT: /srv/tile.openstreetmap.org/html
GATEWAY_INTERFACE: CGI/1.1
HTTPS: on
HTTP_ACCEPT: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9
HTTP_ACCEPT_ENCODING: gzip
HTTP_ACCEPT_LANGUAGE: de,en-US;q=0.9,en;q=0.8
HTTP_CDN_LOOP: Fastly, Fastly
HTTP_FASTLY_CLIENT: 1
HTTP_FASTLY_FF: ovFvuRB7uwtjLzzSaXilgD4YOKcBKgU1q8cIkdUMkC0=!IAD!cache-iad-kiad7000133-IAD, ovFvuRB7uwtjLzzSaXilgD4YOKcBKgU1q8cIkdUMkC0=!IAD!cache-iad-kjyo7100068-IAD
HTTP_FASTLY_ORIG_ACCEPT_ENCODING: gzip, deflate, br
HTTP_FASTLY_SSL: 1
HTTP_HOST: tile.openstreetmap.org
HTTP_SEC_CH_UA: " Not A;Brand";v="99", "Chromium";v="101", "Opera";v="87"
HTTP_SEC_CH_UA_MOBILE: ?0
HTTP_SEC_CH_UA_PLATFORM: "Linux"
HTTP_SEC_FETCH_DEST: document
HTTP_SEC_FETCH_MODE: navigate
HTTP_SEC_FETCH_SITE: none
HTTP_SEC_FETCH_USER: ?1
HTTP_UPGRADE_INSECURE_REQUESTS: 1
HTTP_USER_AGENT: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/101.0.4951.64 Safari/537.36 OPR/87.0.4390.25
HTTP_X_FORWARDED_FOR: 2001:67c:2628:647:12::35f
HTTP_X_FORWARDED_HOST: tile.openstreetmap.org
HTTP_X_FORWARDED_SERVER: cache-iad-kiad7000133-IAD
HTTP_X_TIMER: S1653402280.623487,VS0
HTTP_X_VARNISH: 3111619627, 4007938031
LC_CTYPE: C.UTF-8
PATH: /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin
QUERY_STRING:
REMOTE_ADDR: 2001:67c:2628:647:12::35f
REMOTE_PORT: 2922
REQUEST_METHOD: GET
REQUEST_SCHEME: https
REQUEST_URI: /cgi-bin/debug
SCRIPT_FILENAME: /srv/tile.openstreetmap.org/cgi-bin/debug
SCRIPT_NAME: /cgi-bin/debug
SCRIPT_URI: https://tile.openstreetmap.org/cgi-bin/debug
SCRIPT_URL: /cgi-bin/debug
SERVER_ADDR: 140.211.167.98
SERVER_ADMIN: webmaster@openstreetmap.org
SERVER_NAME: tile.openstreetmap.org
SERVER_PORT: 443
SERVER_PROTOCOL: HTTP/1.1
SERVER_SIGNATURE:

Apache/2.4.41 (Ubuntu) Server at tile.openstreetmap.org Port 443

SERVER_SOFTWARE: Apache/2.4.41 (Ubuntu)
SSL_TLS_SNI: render.openstreetmap.org

**Comment by sommerluk** [comment]
ISP MTN through Opera with built-in VPN:

tile.openstreetmap.org debug
Server Stats
Render Server: pyrene.openstreetmap.org
Load Average: 52.49

File Status
Browser Request Headers
CONTEXT_DOCUMENT_ROOT: /srv/tile.openstreetmap.org/cgi-bin/
CONTEXT_PREFIX: /cgi-bin/
DOCUMENT_ROOT: /srv/tile.openstreetmap.org/html
GATEWAY_INTERFACE: CGI/1.1
HTTPS: on
HTTP_ACCEPT: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9
HTTP_ACCEPT_ENCODING: gzip
HTTP_ACCEPT_LANGUAGE: de,en-US;q=0.9,en;q=0.8
HTTP_CDN_LOOP: Fastly, Fastly
HTTP_FASTLY_CLIENT: 1
HTTP_FASTLY_FF: ovFvuRB7uwtjLzzSaXilgD4YOKcBKgU1q8cIkdUMkC0=!IAD!cache-iad-kiad7000152-IAD, ovFvuRB7uwtjLzzSaXilgD4YOKcBKgU1q8cIkdUMkC0=!IAD!cache-iad-kjyo7100068-IAD
HTTP_FASTLY_ORIG_ACCEPT_ENCODING: gzip, deflate, br
HTTP_FASTLY_SSL: 1
HTTP_HOST: tile.openstreetmap.org
HTTP_SEC_CH_UA: " Not A;Brand";v="99", "Chromium";v="101", "Opera";v="87"
HTTP_SEC_CH_UA_MOBILE: ?0
HTTP_SEC_CH_UA_PLATFORM: "Linux"
HTTP_SEC_FETCH_DEST: document
HTTP_SEC_FETCH_MODE: navigate
HTTP_SEC_FETCH_SITE: none
HTTP_SEC_FETCH_USER: ?1
HTTP_UPGRADE_INSECURE_REQUESTS: 1
HTTP_USER_AGENT: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/101.0.4951.64 Safari/537.36 OPR/87.0.4390.25
HTTP_X_FORWARDED_FOR: 2001:67c:2628:647:12::35f
HTTP_X_FORWARDED_HOST: tile.openstreetmap.org
HTTP_X_FORWARDED_SERVER: cache-iad-kiad7000152-IAD
HTTP_X_TIMER: S1653402238.174376,VS0
HTTP_X_VARNISH: 881833839, 4007505515
LC_CTYPE: C.UTF-8
PATH: /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin
QUERY_STRING:
REMOTE_ADDR: 2001:67c:2628:647:12::35f
REMOTE_PORT: 5700
REQUEST_METHOD: GET
REQUEST_SCHEME: https
REQUEST_URI: /cgi-bin/debug
SCRIPT_FILENAME: /srv/tile.openstreetmap.org/cgi-bin/debug
SCRIPT_NAME: /cgi-bin/debug
SCRIPT_URI: https://tile.openstreetmap.org/cgi-bin/debug
SCRIPT_URL: /cgi-bin/debug
SERVER_ADDR: 140.211.167.98
SERVER_ADMIN: webmaster@openstreetmap.org
SERVER_NAME: tile.openstreetmap.org
SERVER_PORT: 443
SERVER_PROTOCOL: HTTP/1.1
SERVER_SIGNATURE:

Apache/2.4.41 (Ubuntu) Server at tile.openstreetmap.org Port 443

SERVER_SOFTWARE: Apache/2.4.41 (Ubuntu)
SSL_TLS_SNI: render.openstreetmap.org

**Comment by tomhughes** [comment]
So the first one (Orange, no VPN) is coming from an Orange IP and hitting the cache in Toronto and the US render server which is a little surprising but probably reflects how Orange are announcing routes.

The second (MTN, no VPN) is coming from an MTN IP and hitting the cache in London and an Amsterdam render server.

So there's no sign of any common interception and they're using completely different caches and render servers so there shouldn't be anything in common in terms of how up to date things are.

**Comment by pnorman** [comment]
Connection | IP | Cache | Render servers
--|--|--|--
Orange | 160.154.248.187 | YYZ | NA
MTN | 105.235.71.143 | LCY | Europe
VPN | 2001:67c:2628:647:12::35f | IAD | NA

YYZ is an odd routing, but that's up to Fastly and your ISP. I'm not surprised to see sub-optimal routes in Africa, given the tendency towards poor network topology in the region.

If shift-reloading on osm.org does not work, it's not likely to be a CDN issue.

Do you see the path now? Looking at https://tile.openstreetmap.org/17/64062/63579.png I see the footpath rendered, and I'm fetching from a path that hits the NA render server

If https://tile.openstreetmap.org/17/64062/63579.png does not show the path, please open up devtools or your browser's equivalent and provide the full request headers.

To try to obtain some clarity, I queried the CDN logs with and filtered out those not from Linux

```sql
SELECT * FROM fastly_logs_v18
WHERE year=2022 and month=05 and day IN (22,23,24, 25)
and request LIKE 'GET /17/64062/63579.png %' AND referer LIKE 'https://www.openstreetmap.org/%'
```

Source | useragent | asn | country | datacenter | duration | cachehit | render | year | month | day | hour
--|--|--|--|--|--|--|--|--|--|--|--
MTN | FF 100 | 36974 | CI | LON | 2990 | MISS | ysera.openstreetmap.org | 2022 | 5 | 24 | 9
? | FF 91 | 34362 | HR | SOF | 130268 | MISS | culebre.openstreetmap.org | 2022 | 5 | 25 | 18
? | Chrome 101 | 21334 | HU | VIE | 51994 | MISS | culebre.openstreetmap.org | 2022 | 5 | 24 | 11
? | FF 100 | 36974 | CI | LON | 909 | HIT | ysera.openstreetmap.org | 2022 | 5 | 24 | 10

I didn't get any clarity.

**Comment by matkoniecz** [comment]
https://www.openstreetmap.org/note/3194196#map=17/50.90179/16.74531 reported that city is still rendered as platform 

Edit to https://www.openstreetmap.org/relation/13034334/history#map=15/50.8987/16.7437 was made two weeks ago, I tried all ways to refresh tiles

**Comment by pnorman** [comment]
This issue is about problems encountered with viewing from Africa. @matkoniecz, aren't you viewing from Kraków, Poland?

**Comment by matkoniecz** [comment]
Yes, but I thought that it may be relevant and problem may be not-Africa specific. And thanks for mentioning my location, I should add it

**Comment by pnorman** [comment]
> Yes, but I thought that it may be relevant and problem may be not-Africa specific. And thanks for mentioning my location, I should add it

This is what your profile states :P Unfortunately, Europe is the most difficult region to debug anything with, because it hits multiple caches. /status and /dirty may not do what you expect.

I'm getting some oddities with the landuse=residential area in your link, and I'm hitting the US render server and caches. For 15/17908/10984 I triggered a manual dirty by hitting the backend /dirty endpoint, but 15/17908/10983 is in a different metatile so it didn't update. This is expected, but it verifies that the geom is in the DB correctly.

Doing a shift-reload on osm.org and capturing the request gave me the curl

```
curl 'https://tile.openstreetmap.org/15/17908/10983.png' \
  -H 'authority: tile.openstreetmap.org' \
  -H 'accept: image/avif,image/webp,image/apng,image/svg+xml,image/*,*/*;q=0.8' \
  -H 'accept-language: en-CA,en-GB;q=0.9,en-US;q=0.8,en;q=0.7' \
  -H 'cache-control: no-cache' \
  -H 'cookie: _osm_totp_token=655124' \
  -H 'pragma: no-cache' \
  -H 'referer: https://www.openstreetmap.org/' \
  -H 'sec-ch-ua: " Not A;Brand";v="99", "Chromium";v="101", "Google Chrome";v="101"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "Windows"' \
  -H 'sec-fetch-dest: image' \
  -H 'sec-fetch-mode: no-cors' \
  -H 'sec-fetch-site: same-site' \
  -H 'user-agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/101.0.4951.67 Safari/537.36' \
  --verbose -o 1.png \
  --compressed
```

The response was from pyrene via SEA cache, serving a 20568 byte file. Running the same request against pyrene directly gave the same answer.

According to pyrene's /status for the metatile, it is clean and last rendered Mon May 16 19:20:36 2022.

I made a [trivial improvement](https://www.openstreetmap.org/changeset/121500967) with a node move to see if that causes it to be stale. Although this change is one tile over in 15/17907/10983, it should let me know what's going on.

**Comment by matkoniecz** [comment]
it resulted in a rendering update, I guess I should tried this before posting here:

![screen05](https://user-images.githubusercontent.com/899988/170433971-5141386b-250c-4898-b534-8dea49e74ebb.png)

**Comment by tomhughes** [comment]
It's because the change was to a relation and because of their complexity relations are not analysed by the expiry script that marks tiles as dirty.

**Comment by sommerluk** [comment]
> Do you see the path now? Looking at https://tile.openstreetmap.org/17/64062/63579.png I see the footpath rendered, and I'm fetching from a path that hits the NA render server

Yes, I see the path now:
![63579](https://user-images.githubusercontent.com/6830724/170678567-2f3d4813-d184-417e-b626-0287f885cedd.png)

On the other hand, the neighbour tiles seem to be still out-of-date on zoom level z17; the footway is incomplete and the river (which position had been changed) is still not up-to-date on all tiles:
https://www.openstreetmap.org/#map=17/5.36442/-4.04477
Called on 27.05.2022 10:15 UTC with Firefox, after using F5 to reload the page:
![Screenshot](https://user-images.githubusercontent.com/6830724/170680835-f8bcbdf6-96db-4638-9be6-9676404e5240.png)

**Comment by tomhughes** [comment]
The north american render server is quite buy (at least relative to the others) and it will prioritise rendering tiles which don't exist yet over rerendering out of date ones when it is busy.

**Comment by pnorman** [comment]
Closing. There are three issues here.

The first is that the Orange connection is routing to a Toronto, Canada node (YYZ). There's nothing we can do about this, as it is up to Fastly and the ISP. Given the non-ideal internet topology in Africa, Toronto might even be a sane option. Hitting a different cache node will result in higher latency and different backend tiles, but is not itself a cause of outdated tiles.

The second is that the US cache is overloaded, causing dropped tiles. I've moved Fastly traffic hitting the IAD datacenter to the European servers to alleviate some of the load on the US render server and will look at if I should move another node. Long term we need more North American render capacity, but this is not connected to Africa.

The third issue is unrelated, and that's that of the landuse=residential relation tag change not causing re-rendering. This is by design with the change detection algorithm chosen. We could reconsider this design decision, but there are tradeoffs, and if we wanted to do it, it would need its own issue, unrelated to this one.
